src/basic/log.c

```diff
@@ -1478,3 +1478,18 @@ void log_setup(void) {
         if (log_on_console() && show_color < 0)
                 log_show_color(true);
 }
+
+void log_dropped_errors(size_t num_dropped_errors, int level) {
+        if (num_dropped_errors == 0)
+                return;
+
+        log_full(level, "Dropped %zu similar message(s)", num_dropped_errors);
+}
+
+bool log_error_is_ratelimited(int error, LogRateLimit *log_ratelimit) {
```

Comment:
It's very unintuitive that this function is called `log_error_is_ratelimited()`, yet it still modifies the rate limit struct. I don't think this function adds much value, I'd remove it and inline it into the macro.